chapter04/jxp/ch4ex5.pl

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/perl
+use strict;
+use warnings;
+use v5.10;
+
+sub greet {
+  state @previous;
+  my $name = shift;
+  if( scalar @previous == 0 ){
```

Comment:
Just <code> if (scalar @previous) </code> should be enough here. That's just me being picky since everything else looks good :+1: